### PR TITLE
Added test file for istio/istioctl/pkg/authz/analyzer.go

### PR DESCRIPTION
**Pull request by biswajit-9776**
**Please provide a description of this PR:**
Adds unit test coverage

**Please check any characteristics that apply to this pull request.**
Issue #37354 
- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: istioctl/pkg/authz/analyzer_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+						TypeUrl: "type.googleapis.com/envoy.admin.v3.ListenersConfigDump",
+					},
+					LastUpdated: timestamppb.Now(),
+				},
+				ClientStatus: 453,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```
Comment (hanxiaop) [inline istioctl/pkg/authz/analyzer_test.go:96]: I think we've already had a top level test in https://github.com/istio/istio/blob/master/istioctl/pkg/authz/authz_test.go to cover the analyzer, is this testing something that is not covered?

Comment (biswajit-9776) [reply istioctl/pkg/authz/analyzer_test.go:96]: The authz_test.go tests just the checkCmd Cobra command in authz.go file. The checkCmd internally uses NewAnalyzer() and Print() functions, but they do not have their respective unit tests; hence, the analyzer_test.go tries to fulfil it.

Comment (hanxiaop) [reply istioctl/pkg/authz/analyzer_test.go:96]: Great, thanks for the contribution

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**